realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -381,6 +381,34 @@ public int size() {
         }
     }
 
+    /**
+     * Returns true if the list contains the specified element, when attached to a Realm. This
+     * method will query the native Realm core engine to quickly find the specified element.
+     *
+     * If this list is not attached to a Realm, the default {@link List#contains(Object)}
+     * operation will occur.
```

Comment:
If the object is not managed by Realm but list is, {@code false} will be returned.